druid/src/window.rs

```diff
@@ -182,21 +182,26 @@ impl<T: Data> Window<T> {
             );
         }
 
-        if self.root.state().needs_window_origin && !self.root.state().needs_layout {
-            let event = LifeCycle::Internal(InternalLifeCycle::ParentWindowOrigin);
+        if widget_state.children_view_context_changed && !widget_state.needs_layout {
+            let event =
+                LifeCycle::Internal(InternalLifeCycle::RouteViewContextChanged(ViewContext {
+                    window_origin: Point::ORIGIN, // not the same as the root widget's origin
+                    last_mouse_position: self.last_mouse_pos,
+                    clip: self.size.to_rect(),
+                }));
             self.lifecycle(queue, &event, data, env, false);
         }
 
         // Update the disabled state if necessary
         // Always do this before updating the focus-chain
-        if self.root.state().tree_disabled_changed() {
+        if widget_state.tree_disabled_changed() {
```

Comment:
For the record this looks like a bugfix as it has the same *children* tracking logic that view context has. 👍